SALTO-3634: Some fixes to Scriptrunner

IdoZakk opened the pull request:
In script runner multiple values are brought as a single string with '|||' in between, we want to make them arrays
Also added support for email and slack post functions in DC
---
_Release Notes_: 
None

---
_User Notifications_: 
None

alonstern commented on packages/jira-adapter/src/filters/script_runner/workflow_ors.ts:
```diff
@@ -0,0 +1,105 @@
+}
+
+const replaceOr: WalkOnFunc = ({ value }): WALK_NEXT_STEP => {
+  if (typeof value === 'object' && value !== null) {
```
nit: you can do `_.isPlainObject`

IdoZakk commented on packages/jira-adapter/src/filters/script_runner/workflow_ors.ts:
```diff
@@ -0,0 +1,105 @@
+const returnOr: WalkOnFunc = ({ value }): WALK_NEXT_STEP => {
+  if (typeof value === 'object' && value !== null) {
+    Object.entries(value)
+      .filter((entry): entry is [string, string[]] => Array.isArray(entry[1]))
```
Yes, as far as I saw and understand they kept a flat structure (hence the ||| usage). Do you think we should keep a hidden variable to indicate that we did a switch here?

alonstern replied:
After more thinking about it, I think if possible we should have a predefined set of fields in which this will be applied to because.
1. If the user will add this `|||` to a place he is not supposed to (e.g., the script content) we will turn it in to a list
2. If there is a field that supposed to be a list but has only one value it will not be presented as a list (which is less critical but a bit weird)

IdoZakk commented:
* I changed the logic to support only specific fields.
* Single value fields of those specific fields will also be turned into arrays
* I removed all empty fields (they pass deployment without a problem)
* I fixed an error in which 'FIELD_ADDITIONAL_SCRIPT' was not treated as a script

alonstern reviewed (approved):
👌
